src/main/java/no/nav/veilarbaktivitet/db/dao/AktivitetDAO.java

```diff
@@ -66,24 +67,47 @@ public long nesteVersjon() {
     }
 
     @Transactional
-    public void insertAktivitet(AktivitetData aktivitet) {
-        insertAktivitet(aktivitet, new Date());
+    public void oppdaterAktivitet(AktivitetData aktivitet) {
+        oppdaterAktivitet(aktivitet, new Date());
     }
 
     @Transactional
-    public void insertAktivitet(AktivitetData aktivitet, Date endretDato) {
+    public AktivitetData oppdaterAktivitet(AktivitetData aktivitet, Date endretDato) {
         long aktivitetId = aktivitet.getId();
-        final String fhoId = ofNullable(aktivitet.getForhaandsorientering())
-                .map(Forhaandsorientering::getId)
-                .orElse(null);
 
-        SqlParameterSource parms = new MapSqlParameterSource("aktivitet_id", aktivitetId);
+        SqlParameterSource params = new MapSqlParameterSource("aktivitet_id", aktivitetId);
         //language=SQL
-        database.getNamedJdbcTemplate().update("UPDATE AKTIVITET SET gjeldende = 0 where aktivitet_id = :aktivitet_id", parms);
+        List<Long> versjoner = database.getNamedJdbcTemplate().queryForList("SELECT VERSJON FROM AKTIVITET where aktivitet_id = :aktivitet_id FOR UPDATE", params, Long.class);
```

Comment:
her burde det holde og bare hente ut gjeldende, siden den skal matche.
Og alle oppdateringenr vil treffe den.

Reply:
foresten hva skjer hvis mere en en prøver og selecte for update?